integration/spark/src/main/scala/org/apache/spark/sql/index/CarbonIndexUtil.scala

```diff
@@ -299,6 +301,30 @@ object CarbonIndexUtil {
         .Map((carbonLoadModel.getSegmentId, carbonLoadModel.getFactTimeStamp))
     }
     val header = indexTable.getCreateOrderColumn.asScala.map(_.getColName).toArray
+    var segmentList = new ListBuffer[String]()
+    if (isInsertOverWrite) {
+      val loadMetadataDetails = carbonLoadModel.getLoadMetadataDetails.asScala
+      for (loadMetadata <- loadMetadataDetails) {
+        if (loadMetadata.getSegmentStatus != SegmentStatus.INSERT_OVERWRITE_IN_PROGRESS) {
+          segmentList += loadMetadata.getLoadName
+          segmentIdToLoadStartTimeMapping.put(loadMetadata.getLoadName,
+            loadMetadata.getLoadStartTime)
+        }
+      }
+      if (segmentList.nonEmpty) {
```

Comment:
in case of main table load flow, first we make the new segment as Insert_Overwrite_Inprogress, then load that data and at last we make marked for delete for overwritten segment and success for new segment. Same thing please follow for SI also, if you make marked for delete at the beginning, then concurrent clean files and query can cause issues, so once we make the new segment success in SI, then make the marked for delete for all the overwritten segment.

Reply:
made the changes